contracts/EthernalGatesReview.sol

```diff
@@ -0,0 +1,164 @@
+pragma solidity 0.8.11;
+
+/// @author Hammad Ghazi
+contract EthernalGatesReview is ERC721A, Ownable {
```

Comment:
@hammadghazi, you could use [ERC721AQueryable](https://chiru-labs.github.io/ERC721A/#/erc721a-queryable) extension, this project only has 6,000 tokens, and it's a perfect fit for this extension, providing us rooms for futures additional features like soft staking.  
Be aware that you should first update the ERC721A library to `v3.2.0`.

Reply:
done